velox/docs/develop/cache.rst

```diff
@@ -0,0 +1,152 @@
+===========================
+AsyncDataCache (File Cache)
+===========================
+
+Background
+----------
+Velox provides transparent file cache (AsyncDataCache) to accelerate table scan through the hot data reuse and prefetch. The file cache is integrated with the memory system to achieve dynamic memory sharing between file cache and query memory. When a query fails to allocate memory, we retry the allocation by shrinking the file cache. Therefore, the file cache size is automatically adjusted in response to the query memory usage change. See `Memory Management - Velox Documentation <https://facebookincubator.github.io/velox/develop/memory.html>`_  for more information about Velox's file cache.
+
+Configuration Properties
+------------------------
+The AsyncDataCache can be enabled by setting the following config:
+
+.. code-block:: bash
+
+    async-data-cache-enabled=true
+
+
+Other Properties
+----------------
+There is a ``cache.no_retention`` session property in Velox that can be set to control if a query's cached data is retained or not after its execution.
+
+.. list-table::
+   :widths: 30 10 10 70
+   :header-rows: 1
+
+   * - Property Name
+     - Type
+     - Default Value
+     - Description
+   * - cache.no_retention
+     - bool
+     - false
+     - If set to true, evicts out a query scanned data out of in-memory cache right after the access, and also skips staging to the SSD cache.​
```

Comment:
Perhaps better wording:
If set to true, evicts data read by a query (using a table scan) from the in-memory cache right after the access and also skips staging to the SSD cache.​

No comma before "and" needed (no list).